session.go

```diff
@@ -2293,17 +2310,39 @@ func (t *traceWriter) Trace(traceId []byte) {
 	fmt.Fprintf(t.w, "Tracing session %016x (coordinator: %s, duration: %v):\n",
 		traceId, coordinator, time.Duration(duration)*time.Microsecond)
 
-	iter = t.session.control.query(`SELECT event_id, activity, source, source_elapsed, thread
+	for i := 0; i < fetchAttempts; i++ {
+		lastActivity := ""
+		iter := t.session.control.query(`SELECT event_id, activity, source, source_elapsed, thread
 			FROM system_traces.events
 			WHERE session_id = ?`, traceId)
 
-	for iter.Scan(&timestamp, &activity, &source, &elapsed, &thread) {
-		fmt.Fprintf(t.w, "%s: %s [%s] (source: %s, elapsed: %d)\n",
-			timestamp.Format("2006/01/02 15:04:05.999999"), activity, thread, source, elapsed)
-	}
+		for iter.Scan(&timestamp, &activity, &source, &elapsed, &thread) {
+			lastActivity = activity
```

Comment:
This assumption that "Done processing" can play trics on you, let's scan the whole thing.